Bottom: 680265e7759db43f397dcfe95d7e667d6c8b9189
Top:    5fe55c63f76c2825f26076dc341125ff51f90901
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-29 00:21:24 +0100

Refresh of add-a-demonstration-model-for-

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 1886d1c..2169c14 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -916,7 +916,8 @@ hidgl_fill_pcb_polygon (PolygonType *poly, const BoxType *clip_box /*, bool forc
   hidgl_flush_triangles (&buffer);
 
   glPushAttrib (GL_STENCIL_BUFFER_BIT |                 /* Resave the stencil write-mask etc.., and */
-                GL_COLOR_BUFFER_BIT);                   /* the colour buffer write mask etc.. for part way restore */
+                GL_COLOR_BUFFER_BIT |                   /* the colour buffer write mask etc.. for part way restore */
+                GL_DEPTH_BUFFER_BIT);
   glColorMask (0, 0, 0, 0);                             /* Disable writting in color buffer */
   glDepthFunc (GL_ALWAYS);
   glDepthMask (GL_FALSE);
diff --git a/src/hid/gtk/gtkhid-gl.c b/src/hid/gtk/gtkhid-gl.c
index 360fff6..6baad47 100644
--- a/src/hid/gtk/gtkhid-gl.c
+++ b/src/hid/gtk/gtkhid-gl.c
@@ -2013,7 +2013,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     if (!TEST_FLAG (THINDRAWFLAG, PCB)) {
       /* Mask out drilled holes */
       hidgl_flush_triangles (&buffer);
-      glPushAttrib (GL_COLOR_BUFFER_BIT);
+      glPushAttrib (GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
       glColorMask (0, 0, 0, 0);
       glDepthMask (GL_FALSE);
       if (PCB->PinOn) r_search (PCB->Data->pin_tree, drawn_area, NULL, hole_callback, NULL);
